plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -172,10 +190,25 @@ def __init__(self) -> None:
         self._slicing_error_message.actionTriggered.connect(self._reportBackendError)
 
         self._resetLastSliceTimeStats()
-        self._snapshot = None #type: Optional[QImage]
+        self._snapshot: Optional[QImage] = None 
 
         application.initializationFinished.connect(self.initialize)
 
+    def startPlugins(self) -> None:
+        """
+        Ensure that all backend plugins are started
+        It assigns unique ports to each plugin to avoid conflicts.
+        :return:
+        """
+        backend_plugins = CuraApplication.getInstance().getBackendPlugins()
+        for backend_plugin in backend_plugins:
+            if backend_plugin.isRunning():
+                continue
+            # Set the port to prevent plugins from using the same one.
+            backend_plugin.setPort(self._last_backend_plugin_port)
+            self._last_backend_plugin_port += 1
```

Comment:
Maybe limit the ports and ensure that they are within this range: https://stackoverflow.com/questions/10476987/best-tcp-port-number-range-for-internal-applications

We should also maybe query before hand if a port is used already before assigning it.